Lib/contextlib.py

```diff
@@ -102,15 +102,20 @@ def __exit__(self, type, value, traceback):
                 # Suppress StopIteration *unless* it's the same exception that
                 # was passed to throw().  This prevents a StopIteration
                 # raised inside the "with" statement from being suppressed.
-                return exc is not value
+                # Conform to PEP 8 - "Either all return statements in a function
+                # should return an expression, or none of them should".
+                if exc is not value:
+                    return True
+                else:
+                    return False
             except RuntimeError as exc:
                 # Don't re-raise the passed in exception. (issue27122)
                 if exc is value:
                     return False
                 # Likewise, avoid suppressing if a StopIteration exception
                 # was passed to throw() and later wrapped into a RuntimeError
                 # (see PEP 479).
-                if exc.__cause__ is value:
+                if exc.__cause__ is value and type is StopIteration:
```

Comment:
To match the order of the description in the comment, I suggest checking the passed in exception type first:

    if type is StopIteration and exc.__cause__ is value:
        return False